File: ex5/ex5.py
def reversePalindrome(text):

    def isPalindrome(word):
        return word == word[::-1]
    
    words = text.split(" ")
    palindromes = []
    palindromeIndices = []

    for word in words:
        if isPalindrome(word):
            palindromes.append(word)
            palindromeIndices.append(words.index(word))
    
    count = -1
    for index in palindromeIndices:
        words[index] = palindromes[count]
        count = count -1
    
    return " ".join(words)

File: ex5/ex5.js
function reversePalindrome(text) {

    function isPalindrome(word) {
        var re = /[^A-Za-z0-9]/g;
        word = word.toLowerCase().replace(re, '');
        var len = word.length;
        for (var i = 0; i < len / 2; i++) {
            if (word[i] !== word[len - 1 - i]) {
                return false;
            }
        }
        return true;
    }

    words = text.split(" ");

    palindromes = [];
    palindromeIndices = [];

    num_of_words = words.length;

    for(i=0; i<num_of_words; i++) {
        if (isPalindrome(words[i])) {
            palindromes.push(words[i]);
            palindromeIndices.push(i);
        }
    }

    palindromeIndices.forEach(i => {
        words[i] = palindromes.pop()
    });

    return words.join(" ");
}

S1 = "mom and dad went to eye hospital";
S2 = "wow it is next level";

console.log(reversePalindrome(S1));
console.log(reversePalindrome(S2));
